pkg/storage/storagewrappers/sharediterator/shared_iterator_datastore.go

```diff
@@ -135,36 +148,55 @@ func (sf *IteratorDatastore) ReadStartingWithUser(
 
 	if found {
 		keyItem.counter++
+		keyItem.iter.mu.Lock()
```

Comment:
Shouldn't `keyItem` hold the `mu` so you can also increase `.counter` safely?

Reply:
keyItem.counter is protected by sf.internalStorage.mu.